From 2f6db4586810fdb1274d0d47cc0086ca6e28eba6 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <B38613@freescale.com>
Date: Wed, 25 Jul 2012 15:34:54 +0800
Subject: [PATCH 032/281] ENGR00218288 GPS:update GPS to M4.5 version

update GPS to new version, which tested to be much better than
before version in signal and navigating speed.

Signed-off-by: Jianzheng Zhou <B38613@freescale.com>
---
 imx53_smd/init.rc                                                | 9 ++++++---
 imx5x/init.rc                                                    | 1 +
 imx6/etc/init.rc                                                 | 1 +
 imx6/sabresd/init.rc                                             | 9 ++++++---
 .../overlay/frameworks/base/core/res/res/values/config.xml       | 5 +++++
 5 files changed, 19 insertions(+), 6 deletions(-)

diff --git a/imx53_smd/init.rc b/imx53_smd/init.rc
index 0a5c321..3351d16 100755
--- a/imx53_smd/init.rc
+++ b/imx53_smd/init.rc
@@ -7,9 +7,10 @@ on boot
     # Enable Tethering in the Settings
     setprop ro.tether.denied false
 
+    # Set GPS serial and reset GPIO pin
+    setprop ro.kernel.android.gps /dev/athrnmea
+    setprop persist.gps.oacmode @f
     symlink /dev/ttymxc1 /dev/gpsdevice
-    setprop ro.kernel.android.gps ttya
-    #export uart2 and set the default value
     write /sys/class/gpio/export 44
     write /sys/class/gpio/gpio44/direction "out"
     write /sys/class/gpio/gpio44/value 1
@@ -74,10 +75,12 @@ service magd /system/bin/magd
 
 # Orion InG
 service ing /system/etc/gps/ingsvcd -c /system/etc/gps/Orion.ini
+    socket athrkv9988    stream 666 system system
+    socket athrshmsocket stream 666 system system
+    socket athrsupl59991 stream 666 system system
     class late_start
     user root
     group gps
-    disabled
     oneshot
 
 service wpa_supplicant /system/bin/wpa_supplicant \
diff --git a/imx5x/init.rc b/imx5x/init.rc
index 492b31f..c409b88 100644
--- a/imx5x/init.rc
+++ b/imx5x/init.rc
@@ -178,6 +178,7 @@ on post-fs-data
     mkdir /data/data 0771 system system
     mkdir /data/app-private 0771 system system
     mkdir /data/app 0771 system system
+    mkdir /data/app/ing 0771 root root
     mkdir /data/property 0700 root root
     mkdir /data/system 0775 system system
     mkdir /data/system/wpa_supplicant 0771 wifi wifi
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index d319521..dbdf013 100644
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -180,6 +180,7 @@ on post-fs-data
     mkdir /data/data 0771 system system
     mkdir /data/app-private 0771 system system
     mkdir /data/app 0771 system system
+    mkdir /data/app/ing 0771 root root
     mkdir /data/property 0700 root root
     mkdir /data/system 0775 system system
     mkdir /data/system/wpa_supplicant 0771 wifi wifi
diff --git a/imx6/sabresd/init.rc b/imx6/sabresd/init.rc
index 52162c9..bfb9ac1 100755
--- a/imx6/sabresd/init.rc
+++ b/imx6/sabresd/init.rc
@@ -4,9 +4,10 @@ on boot
     # Set permission for IIM node
     symlink /dev/mxs_viim /dev/mxc_mem
 
-    #Set GPS serial and reset GPIO pin
+    # Set GPS serial and reset GPIO pin
+    setprop ro.kernel.android.gps /dev/athrnmea
+    setprop persist.gps.oacmode @f
     symlink /dev/ttymxc2 /dev/gpsdevice
-    setprop ro.kernel.android.gps ttya
     write /sys/class/gpio/export 60
     write /sys/class/gpio/gpio60/direction "out"
     write /sys/class/gpio/gpio60/value 1
@@ -82,10 +83,12 @@ service magd /system/bin/magd
 
 # Orion InG
 service ing /system/etc/gps/ingsvcd -c /system/etc/gps/Orion.ini
+    socket athrkv9988    stream 666 system system
+    socket athrshmsocket stream 666 system system
+    socket athrsupl59991 stream 666 system system
     class late_start
     user root
     group gps
-    disabled
     oneshot
 
 service iprenew_wlan0 /system/bin/dhcpcd -n
diff --git a/imx6/sabresd/overlay/frameworks/base/core/res/res/values/config.xml b/imx6/sabresd/overlay/frameworks/base/core/res/res/values/config.xml
index b263b04..d3b2394 100755
--- a/imx6/sabresd/overlay/frameworks/base/core/res/res/values/config.xml
+++ b/imx6/sabresd/overlay/frameworks/base/core/res/res/values/config.xml
@@ -160,4 +160,9 @@
 
     <!-- Component name of the service providing geocoder API support. -->
     <string name="config_geocodeProvider">com.google.android.location.GeocodeProvider</string>
+
+    <!-- add gpsVendor to support some special code from different gps vendor -->
+    <add-resource type="string" name="config_gpsVendor" />
+    <string name="config_gpsVendor">atheros</string>
+
 </resources>
-- 
1.8.0

